src/app/App.js

```diff
@@ -768,16 +775,16 @@ class App extends EventEmitter {
 	 * Cancels navigation if nextScreen's beforeActivate lifecycle method
 	 * resolves to true.
 	 * @param {!Screen} nextScreen
-	 * @return {!CancellablePromise}
+	 * @return {!Promise}
 	 */
 	maybePreventActivate_(nextScreen) {
-		return CancellablePromise.resolve()
+		return Promise.resolve()
 			.then(() => {
 				return nextScreen.beforeActivate();
 			})
 			.then(prevent => {
 				if (prevent) {
-					this.pendingNavigate = CancellablePromise.reject(new CancellablePromise.CancellationError('Cancelled by next screen'));
+					this.pendingNavigate = Promise.reject(new Error('Cancelled by next screen'));
```

Comment:
So how were these promises previously used. Were any callers explicitly `catch()`-ing `CancellablePromise.CancellationError` instances? Just wondering because we may want to reject with something more specific than just an `Error` (because ultimately this is not actually an "Error" condition, is it, but our use of `Error` makes it look like one).

Reply:
I found these [examples](https://github.com/liferay/senna.js/search?q=catch&unscoped_q=catch).

Reply:
What I meant was, "who is calling `maybePreventActivate_()` and what are they doing with thrown errors?"

- [Here is `maybePreventActivate_()`](https://github.com/liferay/senna.js/blob/257f815c83d865f4da54130df04ca41557307698/src/app/App.js#L785).
- [This](https://github.com/liferay/senna.js/blob/257f815c83d865f4da54130df04ca41557307698/src/app/App.js#L812) is where you construct your `Error` object.
- [This](https://github.com/liferay/senna.js/blob/257f815c83d865f4da54130df04ca41557307698/src/app/App.js#L423) is the sole call-site of `maybePreventActivate_()`.
- And [this](https://github.com/liferay/senna.js/blob/257f815c83d865f4da54130df04ca41557307698/src/app/App.js#L452) is the line where it will eventually receive the `Error`.
- That calls [`handleNavigateError_()`](https://github.com/liferay/senna.js/blob/257f815c83d865f4da54130df04ca41557307698/src/app/App.js#L613).
- Which in turn does a [`console.log` with a stack trace](https://github.com/liferay/senna.js/blob/257f815c83d865f4da54130df04ca41557307698/src/app/App.js#L614) and [emits a "navigationError" event](https://github.com/liferay/senna.js/blob/257f815c83d865f4da54130df04ca41557307698/src/app/App.js#L615) which in theory anybody could be listening to.
- `this.isNavigationPending` will be `false`, so we'll never get into [this first conditional branch](https://github.com/liferay/senna.js/blob/257f815c83d865f4da54130df04ca41557307698/src/app/App.js#L621), but depending on the value of `utils.isCurrentBrowserPath(path)` we still might get down to [this call to `this.removeScreen(path)`](https://github.com/liferay/senna.js/blob/257f815c83d865f4da54130df04ca41557307698/src/app/App.js#L629); I haven't looked at the definition of `isCurrentBrowserPath()`.

So at least some of that seems inappropriate for what is not actually an error condition (eg. logging a stack trace seems overkill, and emitting a "navigationError" as well) — this is just an ordinary consequence of the `beforeActivate` lifecycle method resolving to `true`.

Reply:
@wincent when you have a change, please tell me what you think about the updated code. Thanks ;)